sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -699,6 +699,13 @@ class Column(val expr: Expression) extends Logging {
    */
   def plus(other: Any): Column = this + other
 
+  def plusWithPySparkLoggingInfo(
+      other: Any, loggingInfo: java.util.Map[String, String]): Column = {
+    withOrigin(Some(loggingInfo)) {
+      this + other
+    }
+  }
```

Comment:
Hi, @cloud-fan I added these kind of new `Column` expression methods for taking Python call site information properly along with extending the implementation of the existing `Origin` and `WithOrigin`.

Thanks for your suggestion, now the call site correctness problem seems to be resolved, but I'm wondering if these `{name}PySparkLoggingInfo` methods should be added to all each `Coulmn` APIs.

As this design can be changed after the review, I added for only four functions - `plus`, `divide`, `minus`, and `multiply` - as a prototype for now.

Reply:
This is too much work... Ideally all column creation APIs call `Column.fn` at the end, if we can change pyspark code to only call the Scala `Column.fn`, then we don't need to change the Scala side too much.

Reply:
Sounds good, let me give it a shot. Thanks for sharing the insight!